perfmetrics/scripts/bigquery/bigquery.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2023 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Python module for setting up tables in BigQuery.
+
+This python module creates tables that will store experiment configuration and metrics data in BigQuery.
+
+Note:
+  Make sure BigQuery API is enabled for the project
+"""
+import sys
+from google.cloud import bigquery
+
+class BigQuery():
+
+  configuration_table_id = 'experiment_configuration'
+  fio_table_id = 'read_write_fio_metrics'
+  vm_table_id = 'read_write_vm_metrics'
+  ls_table_id = 'list_metrics'
+
+  def __init__(self, project_id, dataset_id, bq_client=None):
+    if bq_client is None:
+      self.client = bigquery.Client(project=project_id)
+    self.project_id = project_id
+    self.dataset_id = dataset_id
+    dataset = bigquery.Dataset(f"{project_id}.{dataset_id}")
+    self.client.create_dataset(dataset, exists_ok=True)
+    self.dataset_ref = self.client.get_dataset(dataset_id)
+
+  def _execute_query_and_check_error(self, query):
+    """Executes the query in BigQuery.
+
+    Args:
+      query: Query that will be executed in BigQuery.
+
+    Raises:
+      Aborts the program if error is encountered while executing th query.
+    """
+    job = self.client.query(query)
+    job.result()
+    if job.errors:
+      for error in job.errors:
+        print(f"Error message: {error['message']}")
+      sys.exit(1)
+    return
+
+  def setup_bigquery(self):
```

Comment:
Once you add dataset creation logic here, please change it to setup_dataset_and_tables.